### PR TITLE
read and process data from any flat tree

### PR DESCRIPTION
**Pull request by pbuehler**
The functionality was added to read and process data from any flat TTree in a task.
An example is provided by
  Analysis/Tutorials/src/aodwriter.cxx
  Analysis/Tutorials/src/aodreader.cxx

With aodwriter two tables are created and saved to a root file. In aodreader these trees are read and processed.

Usage:
o2-analysistutorial-aodwriter --aod-file AO2D.root --res-file tabletotree > log
o2-analysistutorial-aodreader --aod-file tabletotree_0.root > log

where AO2D.root is an aod file.

### CODE REVIEW

**Comment by ktf** [comment]
Please fix the formatting.

**Comment by pbuehler** [comment]
formatting seems to be ok now

File: Analysis/Tutorials/src/dynamicColumns.cxx [hunk 1]
```diff
@@ -52,14 +52,23 @@ struct BTask {
...
     adaptAnalysisTask<ATask>("produce-track-copy"),
     adaptAnalysisTask<BTask>("check-eta-phi")};
+
+  //  only create table -> tabwle is written to file
+  //return WorkflowSpec{
+  //  adaptAnalysisTask<ATask>("produce-track-copy")};
+
+  // only use table -> table is read from file
+  //return WorkflowSpec{
+  // adaptAnalysisTask<BTask>("check-eta-phi")};
```
Comment (ktf) [inline Analysis/Tutorials/src/dynamicColumns.cxx]: Please do not leave commented code in.

File: Framework/Core/include/Framework/RootTableBuilderHelpers.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #define o2_framework_RootTableBuilderHelpers_H_INCLUDED
 
 #include "Framework/TableBuilder.h"
+#include "Framework/Logger.h"
```
Comment (ktf) [inline Framework/Core/include/Framework/RootTableBuilderHelpers.h]: Why do you need this?

File: Framework/Core/src/DataAllocator.cxx [hunk 3]
```diff
@@ -165,6 +165,42 @@ void DataAllocator::adopt(const Output& spec, TableBuilder* tb)
...
+
+  /// To finalise this we write the table to the buffer.
+  /// FIXME: most likely not a great idea. We should probably write to the buffer
+  ///        directly in the TableBuilder, incrementally.
+  std::shared_ptr<TreeToTable> p(t2t);
+  auto finalizer = [payload = p](std::shared_ptr<FairMQResizableBuffer> b) -> void {
+    auto table = payload->Finalize();
+    LOG(INFO) << "DataAllocator Table created!";
+    LOG(INFO) << "Number of columns " << table->num_columns();
+    LOG(INFO) << "Number of rows    " << table->num_rows();
```
Comment (ktf) [inline Framework/Core/src/DataAllocator.cxx]: Please use `LOG(DEBUG)`.

**Comment by pbuehler** [comment]
I'll clean this up for the next PR

**Pull request merged**